extensions/binding/org.eclipse.smarthome.binding.mqttbroker/src/main/java/org/eclipse/smarthome/binding/mqttbroker/internal/MqttBrokerHandlerFactory.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.mqttbroker.internal;
+
+import java.util.Set;
+
+import org.eclipse.smarthome.binding.mqttbroker.MqttBrokerBindingConstants;
+import org.eclipse.smarthome.binding.mqttbroker.handler.*;
+import org.eclipse.smarthome.core.thing.*;
+import org.eclipse.smarthome.core.thing.binding.*;
+import org.eclipse.smarthome.io.transport.mqtt.MqttService;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.*;
+
+import com.google.common.collect.Sets;
+
+/**
+ * The {@link MqttBrokerHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author David Graeff - Initial contribution
+ */
+@Component(immediate = true, service = ThingHandlerFactory.class, name = "MqttBrokerHandlerFactory")
+public class MqttBrokerHandlerFactory extends BaseThingHandlerFactory {
+    private MqttService mqttService;
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Sets
+            .newHashSet(MqttBrokerBindingConstants.THING_TYPE_TEXTUAL, MqttBrokerBindingConstants.THING_TYPE_BROKER);
```

Comment:
With the Stream API we could remove the dependency to google common here easily:

```Java
Stream.of(MqttBrokerBindingConstants.THING_TYPE_TEXTUAL, MqttBrokerBindingConstants.THING_TYPE_BROKER).collect(Collectors.toSet());
```

Reply:
Could you change the skeleton binding please? That is quite a good solution actually. @martinvw and I were looking for a nice one-liner.

Reply:
The skeleton binding is using a dependency to Google Common? Wuha

Reply:
no, the skeleton uses `java.util.Collections.singleton(THING_TYPE_SAMPLE)`

Reply:
Yup, but so many bindings do it with Guava or Apache libs (the only way in Java7 for a one-liner of course). So having it in the skeleton would be very helpful.

Reply:
Note that we have all means of Java8 available by now - so all uses of Guava/Apache Commons are mainly for historic reasons when we were still supporting Java7.